open-company/open-company-web
# Fix board scroll

### Review after #355 

#### Just to make sure you can check all the test steps
___

This fixes a very annoying bug that was to reset the scroll position of the board when changing location to a post url and back to the board.
It fixes the problem avoiding to have the whole components tree reloaded because it avoids to trigger a url change. It replaces the url with a pushState and replace the internal app router so the components show the post modal.

To test:
- open your browser window enough 
- open a board
- scroll down
- click on a card
- [x] does the post modal appear? Good
- [x] do you NOT see the board changing above and below the post modal while it appeared? Good
- dismiss the modal
- [x] are you back to the board at the same scroll position? Good
- open a digest link
- [x] do you see the post modal correctly populated? Good
- dismiss the post modal
- [x] did you NOT see a spinner in the transition from the modal to AP? Good
- merge
